invenio_records_resources/records/systemfields/files/manager.py

```diff
@@ -248,14 +255,36 @@ def copy(self, src_files, copy_obj=True):
         self.default_preview = src_files.default_preview
         self.order = src_files.order
 
-    def sync(self, src_files):
-        """Sync changes from source files to this manager."""
+    def sync(self, src_files, delete_extras=True):
+        """Sync changes from source files to this manager.
+
+        The source files are fully mirrored with the destination files following the
+        logic:
+
+         * same ObjectVersions are not touched
+         * new ObjectVersions are added to destination
+         * deleted ObjectVersions are deleted in destination
+         * extra ObjectVersions in dest are deleted if `delete_extras` param is
+           True
+        Logic follows the bucket sync logic
+        """
         self.default_preview = src_files.default_preview
         self.order = src_files.order
 
-        # TODO: We don't yet sync file additions/removals/changes from
-        # "src_files". This should take into account if the current bucket
-        # is locked or not.
+        # Sync file additions/removals/changes
+        assert not self.bucket.locked
```

Comment:
wouldn't it be more human friendly to throw an exception with a nice description instead of the assert ?